packages/meteor-wallet/src/lib/meteor-wallet.ts

```diff
@@ -0,0 +1,223 @@
+  params: MeteorWalletParams_Injected,
+  network: Network
+): Promise<MeteorWalletState> => {
+  const keyStore = new keyStores.BrowserLocalStorageKeyStore();
+
+  const near = await connect({
+    keyStore,
+    ...network,
+    headers: {},
+  });
+
+  const wallet = new MeteorWalletSdk({ near, appKeyPrefix: "near_app" });
+
+  // Cleanup up any pending keys (cancelled logins).
+  if (!wallet.isSignedIn()) {
+    await keyStore.clear();
+  }
+
+  return {
+    wallet,
+    keyStore,
+  };
+};
+
+const createMeteorWalletInjected: WalletBehaviourFactory<
+  InjectedWallet,
+  { params: MeteorWalletParams_Injected }
+> = async ({ options, logger, store, params }) => {
+  const _state = await setupWalletState(params, options.network);
+
+  const cleanup = () => {
+    _state.keyStore.clear();
+  };
+
+  const getAccounts = () => {
+    const accountId = _state.wallet.getAccountId();
+
+    if (!accountId) {
+      return [];
+    }
+
+    return [{ accountId }];
+  };
+
+  const transformTransactions = async (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    const account = _state.wallet.account()!;
+    const { networkId, signer, provider } = account.connection;
+
+    const localKey = await signer.getPublicKey(account.accountId, networkId);
+
+    return Promise.all(
+      transactions.map(async (transaction, index) => {
+        const actions = transaction.actions.map((action) =>
+          createAction(action)
+        );
+        const accessKey = await account.accessKeyForTransaction(
+          transaction.receiverId,
+          actions,
+          localKey
+        );
+
+        if (!accessKey) {
+          throw new Error(
+            `Failed to find matching key for transaction sent to ${transaction.receiverId}`
+          );
+        }
+
+        const block = await provider.block({ finality: "final" });
+
+        return nearTransactions.createTransaction(
+          account.accountId,
+          utils.PublicKey.from(accessKey.public_key),
+          transaction.receiverId,
+          accessKey.access_key.nonce + index + 1,
+          actions,
+          utils.serialize.base_decode(block.header.hash)
+        );
+      })
+    );
+  };
+
+  return {
+    async signIn({ contractId, methodNames }) {
+      logger.log("MeteorWallet:signIn", {
+        contractId,
+        methodNames,
+      });
+
+      let signInRequest: IOMeteorWalletSdk_RequestSignIn_Inputs;
+
+      if (notNullEmptyArray(methodNames)) {
+        signInRequest = {
+          methods: methodNames,
+          type: EMeteorWalletSignInType.SELECTED_METHODS,
+          contract_id: contractId,
+        } as IOMeteorWalletSdk_RequestSignIn_Inputs;
+      } else {
+        signInRequest = {
+          type: EMeteorWalletSignInType.ALL_METHODS,
+          contract_id: contractId,
+        } as IOMeteorWalletSdk_RequestSignIn_Inputs;
+      }
+
+      await _state.wallet.requestSignIn(signInRequest);
+      return getAccounts();
+    },
```

Comment:
We should be able to avoid the variable and type imports by approaching this slightly differently (though I must admit, I'm not sure I understand why methods can't be optional - removing the need for a `type` property):

```suggestion
    async signIn({ contractId, methodNames = [] }) {
      logger.log("MeteorWallet:signIn", {
        contractId,
        methodNames,
      });

      if (methodNames.length) {
        await _state.wallet.requestSignIn({
          methods: methodNames,
          type: EMeteorWalletSignInType.SELECTED_METHODS,
          contract_id: contractId,
        });
      } else {
        await _state.wallet.requestSignIn({
          type: EMeteorWalletSignInType.ALL_METHODS,
          contract_id: contractId,
        });
      }
  
      return getAccounts();
    },
```

Reply:
The reason for the `type` property is just to be more deterministic instead of relying purely on type checks- feels a little strange to me that an empty array equals "all methods", where it feels like it should mean "no methods" (even though that's not an option currently). I'm not sure what cases might be added in the future as well.

Reply:
Also, in the future there might be a `MeteorWalletSignInType.FULL_ACCESS` way to sign in.